earth2mip/networks/graphcast.py

```diff
@@ -59,504 +66,578 @@
+        target_lead_times=slice("6h", f"{eval_steps*6}h"),
+        **dataclasses.asdict(task_config),
+    )
 
-    size = os.path.getsize(checkpoint_path)
-    logging.info(f"Checkpoint Size in MB: {size / 1e6}")
+    print("All Examples:  ", example_batch.dims.mapping)
+    print("Eval Inputs:   ", eval_inputs.dims.mapping)
+    print("Eval Targets:  ", eval_targets.dims.mapping)
+    print("Eval Forcings: ", eval_forcings.dims.mapping)
+
+    # run autoregression
+    assert model_config.resolution in (0, 360.0 / eval_inputs.sizes["lon"]), (
+        "Model resolution doesn't match the data resolution. You likely want to "
+        "re-filter the dataset list, and download the correct data."
+    )
 
-    static_variables = get_static_data(package, resolution)
+    print("Inputs:  ", eval_inputs.dims.mapping)
+    print("Targets: ", eval_targets.dims.mapping)
+    print("Forcings:", eval_forcings.dims.mapping)
 
     # load stats
-    with open(join("stats/diffs_stddev_by_level.nc"), "rb") as f:
+    with open(os.path.join(stats_dir, "diffs_stddev_by_level.nc"), "rb") as f:
         diffs_stddev_by_level = xarray.load_dataset(f).compute()
-    with open(join("stats/mean_by_level.nc"), "rb") as f:
+    with open(os.path.join(stats_dir, "mean_by_level.nc"), "rb") as f:
         mean_by_level = xarray.load_dataset(f).compute()
-    with open(join("stats/stddev_by_level.nc"), "rb") as f:
+    with open(os.path.join(stats_dir, "stddev_by_level.nc"), "rb") as f:
         stddev_by_level = xarray.load_dataset(f).compute()
 
-    # select needed channels from stats
-    in_codes, t_codes = get_codes_from_task_config(task_config)
-    mean = np.array(
-        [get_data_for_code_scalar(code, mean_by_level) for code in in_codes]
-    )
-    scale = np.array(
-        [get_data_for_code_scalar(code, stddev_by_level) for code in in_codes]
-    )
-    target_scale = np.array(
-        [get_data_for_code_scalar(code, diffs_stddev_by_level) for code in t_codes]
-    )
+    # jit the stuff
+    # @title Build jitted functions, and possibly initialize random weights
+    def construct_wrapped_graphcast(
+        model_config: graphcast.ModelConfig, task_config: graphcast.TaskConfig
+    ):
+        """Constructs and wraps the GraphCast Predictor."""
+        # Deeper one-step predictor.
+        predictor = CachedGraphcast(model_config, task_config)
+
+        # Modify inputs/outputs to `graphcast.GraphCast` to handle conversion to
+        # from/to float32 to/from BFloat16
+        predictor = casting.Bfloat16Cast(predictor)
+
+        # Modify inputs/outputs to `casting.Bfloat16Cast` so the casting to/from
+        # BFloat16 happens after applying normalization to the inputs/targets.
+        predictor = normalization.InputsAndResiduals(
+            predictor,
+            diffs_stddev_by_level=diffs_stddev_by_level,
+            mean_by_level=mean_by_level,
+            stddev_by_level=stddev_by_level,
+        )
 
-    return GraphcastTimeLoop(
```

Comment:
`GraphcastTimeLoop` is called in `test/test_graphcast.py::test_load_time_loop` I am removing it from there